### ISSUE TITLE
Connecting extension to alerting server not working

### ISSUE THREAD

**Issue by veeral-patel** [palantir/phishcatch#25]
Here's my debug configuration:

<img width="775" alt="Screen Shot 2021-06-17 at 6 42 47 PM" src="https://user-images.githubusercontent.com/12554095/122493114-fad3d800-cf9b-11eb-9318-93e1e7042076.png">

As you can see I've set the phishcatch server URL to a ngrok site which points to localhost:8000. I am running the server locally at port 8000 using `PRESHARED_KEY=MYPSK dev.sh`

I am able to trigger phishcatch browser notifications, however I am not seeing any alerts logged by my alert server.

What do I need to do to get the extension and server to cooperate?

**Comment by carbureted** [comment]
If you run the server locally (ie without ngrok) do you have the same problem? I'll try to reproduce this tomorrow if you're still having problems.

**Comment by mathsugan** [comment]
I confirm it can't connect locally.

Either with an IP (127.0.0.1), a domain, by using http or https.

**Comment by veeral-patel** [comment]
@carbureted neither local or with ngrok work

**Comment by carbureted** [comment]
You guys are running via docker, right?

**Comment by mathsugan** [comment]
Yes, here is how I launched it :

`docker run -d -e PRESHARED_KEY=justtesting --name phishcatch -p 8080:80 phishcatch`

**Comment by 0xMirasio** [comment]
veeral-patel, it's normal the /alert POST isn't working
There is a CSP policy that isn't allowing to make request to url beginning with http:// 

the csp policy is :   "content_security_policy": "script-src 'self'; object-src 'self'; connect-src https://*",
If you want to your server ngrok to work, you need to add http:// in the CSP policy in manifest.json (/public)
or use the https URL gived by ngrok when requesting a http server. 

That issue can be verified by watching background on chrome dev console 
![Capture du 2021-07-05 16-01-16](https://user-images.githubusercontent.com/36262288/124482724-1f4ff280-dd99-11eb-8786-f089504f5b45.png)

**Comment by carbureted** [comment]
Good catch. I'll update the wiki to note the HTTPS requirement. Do people feel having an https server running is too high of a barrier for initial development?

**Comment by 0xMirasio** [comment]
Well that's not a problem, you can change manifest.json to support http : 

  "content_security_policy": "script-src 'self'; object-src 'self'; connect-src **http://*** https://*",

**Comment by carbureted** [comment]
I could, but it enables hosting prod servers on http. A compromise might be enabling http for all localhost ports.

**Comment by veeral-patel** [comment]
@carbureted hey -- yes, i think that's a good compromise as well
